### PR TITLE
Add Loading State to ButtonSubmit where it's needed.

### LINKED ISSUE #868: Add Loading ButtonSubmit everywhere it's needed
### Problem

With #854 merged we have a new `ButtonSubmit` component thanks to @sandrina-p that turns into a loading symbol on form submission of asynchronous actions.

#854 added this component in some places but not everywhere. For example, it doesn't exist on the voting buttons for voting on proposals.

### Solution

Figure out, on a case-by-case basis, how this button should be integrated in all remaining submit forms where asynchronous network-based actions take place.

### PR DESCRIPTION
**Pull request by sandrina-p**
Closes #868 

The only missing place was when voting in a proposal (yes, no and cancel). If you notice any other place, let me know and I'll add it in this PR.

image

#### Scouting stuff:
- Fix the banner text styles (turn them bold).
- Fix the spinning colors based on the button variant.

#### 🔥How to force an error to appear?
Well, our code is so good that the only way is to make it on purpose by breaking the code 😅.
Go to the file `ProposalVoteOptions.vue`. You'll find 3 functions, each one for each possible action: `voteFor`, `voteAgainst` and `cancelProposal`. 

Each one has a `try { }` block. You need to force an error inside it.

Example:
```js
async cancelProposal () {
  // ...
  try {
     // Add these 2 lines to force the error 2.5s after you click the button.
     await new Promise((resolve, reject) => { setTimeout(resolve, 2500) })
     this.dumb.error = 'forced-dumb-error'
    
     this.refVoteMsg.clean()
     // ...

### CODE REVIEW

File: frontend/views/components/ButtonSubmit.vue [hunk 1]
```diff
@@ -16,12 +16,12 @@
...
 button-submit(
-@click='handleLogin'
+  @click='handleLogin'
 ) Login
 
-This will display a "loading state" on the button while the login is happening.
+This will display a spinner (the loading state) on the button while the login is happening.
 
-The button has type="submit" to catch any way of form submittion (ex: press Enter).
+The button has type="submit" to catch any way of form submission (ex: press Enter).
```
Comment (taoeffect) [inline frontend/views/components/ButtonSubmit.vue:24]: Nice typo find!

File: frontend/views/containers/proposals/ProposalItem.vue [hunk 2]
```diff
@@ -21,6 +21,8 @@
         v-if='proposal.status === statuses.STATUS_OPEN'
         :proposalHash='proposalHash'
       )
+    // Note: $refs.voteMsg is used by children ProposalVoteOptions
+    banner-scoped(ref='voteMsg' data-test='voteMsg')
```
Comment (taoeffect) [inline frontend/views/containers/proposals/ProposalItem.vue:25]: Well done commenting this, as this certainly isn't intuitive! 👍

File: frontend/views/containers/proposals/ProposalVoteOptions.vue [hunk 3]
```diff
@@ -35,18 +30,21 @@ import { VOTE_FOR, VOTE_AGAINST } from '@model/contracts/voting/rules.js'
...
   },
+  components: {
+    ButtonSubmit
+  },
   data () {
     return {
       ephemeral: {
         changingVote: false
-      },
-      errorMsg: null
```
Comment (taoeffect) [inline frontend/views/containers/proposals/ProposalVoteOptions.vue:49]: Excellent 👍

**Review by taoeffect — APPROVED** [review]
@sandrina-p A perfect PR on your first try. Fantastic. :)

**Review by mmbotelho — APPROVED** [review]
Approved @sandrina-p ! Good job 😄

**Pull request merged**